openfoodforest/openfoodforest
# Use apollo server

PR https://github.com/openfoodforest/openfoodforest/pull/1 should be reviewed and merged before this one, because this PR depends on changes in that PR (the addition of the README). 

This patch adds a basic implementation of Apollo Server by just following the tutorial at https://www.apollographql.com/docs/apollo-server/getting-started .

To test:
1. Check out this PR
2. Run `npm install`
3. Run `DEBUG=* node index.js` (Mac) or `set DEBUG=* && node index.js` (Windows)
4. In a browser, go to http://localhost:4000 and check that you wind up on something called the graphql playground.
5. Where it says "write your query or mutation here", try pasting this and pressing the play button:
```
{
   books {
    title
  }
}
```

It should look like this:

<img width="1210" alt="Screen Shot 2020-08-18 at 7 57 09 AM" src="https://user-images.githubusercontent.com/169674/90529320-7076d980-e128-11ea-9cc6-43618c368d7a.png">

If the above all works for you, you can approve and merge this PR. (Don't worry about the code diff right now -- it looks like a mess because I made a mistake in the original setup.) 

Apollo Server and Apollo Client are two halves of an implementation of GraphQL. It will make it easier for us to exchange information between the server and client. In other words, this will take care of a lot of the nitpicky parts of the "talking to" between the client (blue part) and server (yellow part) for us, so we don't have to worry about it ourselves. 

<img width="880" alt="Screen_Shot_2020-08-18_at_7 35 20_AM" src="https://user-images.githubusercontent.com/169674/90528240-41ac3380-e127-11ea-938f-e79b107e90de.png">
